src/zenml/models/v2/base/scoped.py

```diff
@@ -583,3 +587,131 @@ def apply_sorting(
             return query
 
         return super().apply_sorting(query=query, table=table)
+
+
+class RunMetadataAttachedFilter(BaseFilter):
```

Comment:
I prefer something like `RunMetadataFilterMixin`, but feel free to ignore and keep as is.

Reply:
Not a bad idea, renamed it accordingly.